include/tuvx/linalg/linalg.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+namespace tuvx {
+
+/// @struct Tridiagonal Matrix Data Structure.
+/// @brief Given a matrix of shape (n, n), the
+/// data structure holds three std::vectors
+/// storing the diagonal values.
+template <typename T> struct TridiagonalMatrix {
+  std::size_t size;
+  std::vector<T> upper_diagonal_; // upper diagonal
+  std::vector<T> lower_diagonal_; // lower diagonal
+  std::vector<T> main_diagonal_;  // main diagonal
+};
+
+/// @brief Allocates a vector to a given size and fills it with random values
+/// @param x Vector to allocate and fill
+/// @param size Size of the vector
+template <typename T> void FillRandVec(std::vector<T> &x, std::size_t size);
```

Comment:
```suggestion
template <typename T> void FillRandomVector(std::vector<T> &x, std::size_t size);
```
formatting looks good - just want to use full words in names

Reply:
Done! Let me know if I missed anything.